submissions/Shrai-dev/memory-pair-game/main.js

```diff
@@ -0,0 +1,222 @@
+const cardArray = [
+  {
+    id: 1,
+    name: "Cheerful Santa",
+    src: "./assets/cheerful-santa.jpg",
+  },
+  {
+    id: 1,
+    name: "Cheerful Santa",
+    src: "./assets/cheerful-santa.jpg",
+  },
+  {
+    id: 2,
+    name: "Christmas tree",
+    src: "./assets/christmas-tree.jpg",
+  },
+  {
+    id: 2,
+    name: "Christmas tree",
+    src: "./assets/christmas-tree.jpg",
+  },
+  {
+    id: 3,
+    name: "Christmas stocking",
+    src: "./assets/christmas-stocking.jpg",
+  },
+  {
+    id: 3,
+    name: "Christmas stocking",
+    src: "./assets/christmas-stocking.jpg",
+  },
+  {
+    id: 4,
+    name: "Elf",
+    src: "./assets/elf.jpg",
+  },
+  {
+    id: 4,
+    name: "Elf",
+    src: "./assets/elf.jpg",
+  },
+  {
+    id: 5,
+    name: "Snowman",
+    src: "./assets/snowman.jpg",
+  },
+  {
+    id: 5,
+    name: "Snowman",
+    src: "./assets/snowman.jpg",
+  },
+  {
+    id: 6,
+    name: "Gift",
+    src: "./assets/gift.png",
+  },
+  {
+    id: 6,
+    name: "Gift",
+    src: "./assets/gift.png",
+  },
+  {
+    id: 7,
+    name: "Reindeer",
+    src: "./assets/reindeer.jpg",
+  },
+  {
+    id: 7,
+    name: "Reindeer",
+    src: "./assets/reindeer.jpg",
+  },
+  {
+    id: 8,
+    name: "Christmas sleigh",
+    src: "./assets/christmas-sleigh.png",
+  },
+  {
+    id: 8,
+    name: "Christmas sleigh",
+    src: "./assets/christmas-sleigh.png",
+  },
+];
+
+const main = document.querySelector(".main");
+
+const board = document.createElement("div");
+board.classList.add("memory-board");
+main.appendChild(board);
+
+const startScreen = document.createElement("div");
+startScreen.classList.add("overlay-text", "visible");
+startScreen.innerText = "Click to Start";
+main.appendChild(startScreen);
+
+const victoryScreen = document.createElement("div");
+victoryScreen.classList.add("overlay-text");
+victoryScreen.innerText = "Victory";
+victoryScreen.setAttribute("id", "victory-text");
+main.appendChild(victoryScreen);
+
+const restart = document.createElement("span");
+restart.classList.add("overlay-text-small");
+restart.innerText = "Click to Restart";
+victoryScreen.appendChild(restart);
+
+let backImg;
+let frontImg;
+let boardCard;
+let cards;
+let cardsWon = 0;
+
+let hasFlippedCard = false;
+let lockBoard = false;
+let firstCard;
+let secondCard;
+
+document.addEventListener("DOMContentLoaded", function () {
+  startGame();
+
+  function startGame() {
+    createCard(board, cardArray);
+    sortCards();
+    let overlays = Array.from(document.getElementsByClassName("overlay-text"));
+    overlays.forEach((overlay) => {
+      overlay.addEventListener("click", () => {
+        overlay.classList.remove("visible");
+      });
+    });
+
+    cards = Array.from(document.querySelectorAll(".board-card"));
+    cards.forEach((card) => card.addEventListener("click", flipCard));
+  }
+
+  function createCard(boardCard, cardArray) {
```

Comment:
It seems that you are modifying function param. Maybe is would be better to return new object?